src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,247 @@
+	*/
+	private static String logMessage;
+	
+	/**
+	 * Getting handler of the UI module to send the data
+	 */
+	private static IContentNotificationHandler handler;
+	 
+	/**
+	 * @param String message - JSON string message (Meta fields are newUser or message or userExit)
+	 *
+	 * if meta field is newUser, then remaining field of parameter message would only be imageMap (it is a string of json array whose elements are json object having fields username and corresponding image). 
+	 *		onNewUserJoined(String message) - message fields are username and image  
+	 * if meta field is message, then remaining fields of parameter message would be username, message and time. onMessageReceived(String message) - message fields are username, message, time and image
+	 * if meta field is exitUser, then remaining field of parameter message would only be username. onUserExit(String message) - message fields are username and image
+	 * Networking module will be calling this method with json string message to send data to the content module
+	 *
+	 * creating error log message if message in not converting into json object,
+	 * or if not able extract field from object, or if not able to extract handler from hashmap (handlerMap)
+	 * if message will be successfully sent to UI, then creating success log message
+	 */
+	@override
+	public void onMessageReceived(String message) {
+		logMessage = "onMessageReceived method of NetworkMessageHandler class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		
+		try {
+			JSONObject jsonObject = new JSONObject(message);
+		} 
+		catch(Exception e) {
+			logMessage = "converting JSON string to JSON object process failed";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			metafield = jsonObject.getString("meta");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get metafield value of argument message";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		try {
+			handler = handlerMap.get("UI");
+		}
+		catch(Exception e) {
+			logMessage = "failed to get IContentNotification handler from handlerMap";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		
+		jsonObject.remove("meta");
+		
+		if(metafield.equals("newUser")) {
+			logMessage = "New User case";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+			
+			imagemap = new HashMap<String, String>();
+			jsonObjectArray = new JSONArray();
+			
+			try {
+				jsonObjectArray = jsonObject.getJSONArray("imageMap");
+			}
+			catch(Exception e) {
+				logMessage = "failed to get imageMap field value of argument message";
+				logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+				return;
+			}
+			
+			jsonObject.remove("imageMap");
+			for(int i = 0; i < jsonObjectArray.size(); i++) {
+				try {
+					object = jsonObjectArray.getJSONArray(i);
```

Comment:
Required type: `JSONObject`
Provided type: `JSONArray`

Reply:
https://stackoverflow.com/questions/7193599/how-can-i-turn-a-jsonarray-into-a-jsonobject/7193753

LHS is a JSONArray, RHS is a JSONObject. Please refer to the above link